### ISSUE TITLE
Unable to create conan package locally using command "conan create . --build=missing"

### ISSUE THREAD

**Issue by pranavishere2** [eclipse-uprotocol/up-cpp#100]
I am using my personal computer and trying to create conan package locally using command "conan create . --build=missing". I have installed all the local dependencies needed and seeing following error:

Steps followed:
Step 1: Clone up-cpp repository with tag v0.1.2-dev tag
Step 2: Make sure all teh needed dependencies are installed
Step 3: run command "git submodule update --init --recursive"
Step 4: Try to create conan package locally using "conan create . --build=missing"

Error message after trying to create conan package locally:

up-cpp/0.1.2-dev: Running CMake.build()
up-cpp/0.1.2-dev: RUN: cmake --build "/home/pranav/.conan2/p/b/up-cp8f4152f642d08/b" -- -j4
[  5%] Running cpp protocol buffer compiler on /home/pranav/.conan2/p/b/up-cp8f4152f642d08/b/up-core-api/uprotocol//core/usubscription/v3/usubscription.proto
[  5%] Running cpp protocol buffer compiler on /home/pranav/.conan2/p/b/up-cp8f4152f642d08/b/up-core-api/uprotocol//core/udiscovery/v3/udiscovery.proto
[  7%] Running cpp protocol buffer compiler on /home/pranav/.conan2/p/b/up-cp8f4152f642d08/b/up-core-api/uprotocol//core/utwin/v2/utwin.proto
[ 10%] Running cpp protocol buffer compiler on /home/pranav/.conan2/p/b/up-cp8f4152f642d08/b/up-core-api/uprotocol//file.proto
file.proto: This file contains proto3 optional fields, but --experimental_allow_proto3_optional was not set.
gmake[2]: *** [CMakeFiles/up-core-api-protos.dir/build.make:96: up-core-api/file.pb.h] Error 1
gmake[2]: *** Waiting for unfinished jobs....
core/usubscription/v3/usubscription.proto: This file contains proto3 optional fields, but --experimental_allow_proto3_optional was not set.
gmake[2]: *** [CMakeFiles/up-core-api-protos.dir/build.make:80: up-core-api/core/usubscription/v3/usubscription.pb.h] Error 1
core/udiscovery/v3/udiscovery.proto: This file contains proto3 optional fields, but --experimental_allow_proto3_optional was not set.
gmake[2]: *** [CMakeFiles/up-core-api-protos.dir/build.make:72: up-core-api/core/udiscovery/v3/udiscovery.pb.h] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/up-core-api-protos.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2

up-cpp/0.1.2-dev: ERROR: 
Package 'f2ce90f03731b43cf950d45aec8e9ed44989f0bd' build failed
up-cpp/0.1.2-dev: WARN: Build folder /home/pranav/.conan2/p/b/up-cp8f4152f642d08/b
ERROR: up-cpp/0.1.2-dev: Error in build() method, line 54
	cmake.build()
	ConanException: Error 2 while executing

**Comment by PLeVasseur** [comment]
@pranavishere2 -- thanks for your feedback. I will look into it :+1: 

For some further details --
1. Were you able to build a previous tagged release of `up-cpp`?
2. When you mention that you've installed all local dependencies, I _think_ the local dependencies should all be installed for `up-cpp` via Conan, right? Did you install any dependencies of `up-cpp` outside of Conan?

**Comment by pranavishere2** [comment]
**Were you able to build a previous tagged release of up-cpp?**
I have not tried previous release. I tried specifically v0.1.2-dev tag . I will try previous releases and will let you know my findings.

When I mentioned I installed all local dependencies,  I meant say I installed fresh ubuntu 22.04 on my personal laptop with following requirements.
Compiler: GCC/G++ 11 or Clang 13
Conan : 1.59 or latest 2.X

All other dependencies should be installed via conan.

**Comment by PLeVasseur** [comment]
Hi @pranavishere2 -- I was able to build based on a new clone checked out on the `v0.1.2-dev` tag.

Here's the prereqs I had installed:
```bash
$ gcc --version
gcc (Ubuntu 11.4.0-2ubuntu1~20.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
```bash
$ conan --version
Conan version 2.1.0
```

I then followed the README.md:
```bash
git submodule update --init --recursive
```
and
```bash
conan create . --build=missing
```

Which left me with:
```bash
$ conan list up-cpp
Found 1 pkg/version recipes matching up-cpp in local cache
Local Cache
  up-cpp
    up-cpp/0.1.2-dev
```

Could you:
```bash
cd up-cpp/up-core-api
git log
```
and let me know which commit `up-core-api` is checked out on?

It should be:
```
commit 926efe294759b49fbfabbe6713ecab9ef39bbe50 (HEAD, tag: uprotocol-core-api-1.5.7)
```

**Comment by pranavishere2** [comment]
This is what I see in the git log of up-core-api
commit 494b3313b55aaefe5b8b0376dd761b75fa96a78a (HEAD, tag: v0.1.2-dev, origin/main, origin/HEAD, main)
Author: Steven Hartley <steven.hartley@gm.com>
Date:   Sat Apr 20 13:04:01 2024 -0400

    Update version to 0.1.2-dev (#98)
    
    Updating the version and creating a tag

**Comment by PLeVasseur** [comment]
Hey @pranavishere2

I'm a bit confused. What you've posted looks like you ran `git log` from the `up-cpp` folder.

Could you post the `git log` from the `up-cpp/up-core-api` folder?

**Comment by pranavishere2** [comment]
pranav@pranav-Inspiron-5547:~/SDV/up-cpp/up-core-api$ git log -3
commit 926efe294759b49fbfabbe6713ecab9ef39bbe50 (HEAD, tag: uprotocol-core-api-1.5.7)
Author: Steven Hartley <steven.hartley@gm.com>
Date:   Wed Apr 3 14:38:03 2024 -0400

    Urgent fix in uTwin package Definition (#114)
    
    uTwin was changed from v1 to v2 because we were going to call the original uTwin API that used cloudevents as V1 however the package name was not fixed in the proto causing build issues.  This fix addresses that problem as well as a minor document change in uprotocol_options.proto for the topic ID number.

commit 8d1da1e8e92b74ac84a774357b5089fb29cf1cf1
Author: Steven Hartley <steven.hartley@gm.com>
Date:   Tue Apr 2 08:27:29 2024 -0400

    Marking the branch READ-ONLY (#113)
    
    The following change will change the main readme to state that the protos in the core-api have moved to up-spec so that the specifications and core apis can be released and labeled together.  Any future changes to the core APIs need to be made on the up-spec project.

commit bdaf73791fce1581ff6b3f0ee6b29ed805473d8f
Author: eclipse-uprotocol-bot <uprotocol-bot@eclipse.org>
Date:   Fri Mar 15 01:16:06 2024 +0000

    [maven-release-plugin] prepare for next development iteration

**Comment by gregmedd** [comment]
Can this issue be closed?
